### ISSUE TITLE
Awair Temp displayed in Celsius in Home (other accessories display in Fahrenheit as expected)

### ISSUE THREAD

**Issue by jm3** [DMBlakeley/homebridge-awair2#514]
**Describe Your Problem:**

I am in the US where Fahrenheit is the expected default.

The Awair Sensor, via the Homebridge plugin, reports in what looks like Celsius — 32º — not Fahrenheit. There's no setting in the plugin to change this.

My other Homebridge accessories —e.g. smart thermostat, car — display their temp using Fahrenheit, either via an explicit setting, or automatically.

FWIW, I have the UI Setting in Homebridge to Fahrenheit as well.

**Logs:**

```
Show the Homebridge logs here, remove any sensitive information.
```

**Plugin Config:**

```json
{
  "token": "DEADBEEF.CAFEBABE.DECAFBAD",
  "userType": "users/self",
  "airQualityMethod": "awair-aqi",
  "endpoint": "15-min-avg",
  "limit": 1,
  "carbonDioxideThreshold": 1000,
  "carbonDioxideThresholdOff": 800,
  "occupancyDetection": false,
  "occupancyRestart": false,
  "logging": false,
  "verbose": false,
  "development": false,
  "platform": "Awair2"
},
```

**Screenshots:**

**Environment:**

* **Plugin Version**: homebridge-awair2 v5.10.4
* **Homebridge Version**: v1.6.0
* **Node.js Version**: v18.13.0
* **NPM Version**: 8.19.3
* **Operating System**: hb-service / Synology NAS -->

- [x] Homebridge UI Settings are set to Fahrenheit
- [x] iPhone and Mac are set to US region settings (i.e. everything else displays in Fahrenheit)
- [x] Latest plugin version
- [x] Latest Homebridge version

My other Homebridge accessories —e.g. smart thermostat, car — display their temp using Fahrenheit, either via an explicit setting, or automatically.

<img width="507" alt="Home: Accessory settings showing 30º" src="https://user-images.githubusercontent.com/12213/233867035-47e464db-349b-4507-8df3-a60f4808d606.png">

<img width="263" alt="Home summary showing 30º" src="https://user-images.githubusercontent.com/12213/233867045-3909de3d-04bc-4bc8-a43a-c97ccd0d4224.png">

**Comment by DMBlakeley** [comment]
Hi @jm3,

Thanks for reaching out. I am also in the U.S. and temperatures are displayed in F according to Homebridge and HomeKit settings. The Awair plugin does not have a setting to override.

What I find unusual is the temperature of 32 degrees and 0% humidity. 32 degrees is possible but 0% humidity is highly unlikely. Seems to indicate that your data is not being read from the Awair servers. Would you first check that the Awair device is being read correctly by the Awair app on your iPhone. If yes, could you please turn on "logging" and "verbose" and share the initial boot logs with me? There may be some information which you will want to redact before sharing.

Thanks,
Doug

**Comment by jm3** [comment]
Good point, you're right — the zero 0% humidity is a good indicator that things aren't what they seem from the Homebridge plugin's POV.

Will take me a minute to grab the requested verbose logs.

In the meantime, fwiw, the actual Awair hardware device reports measurements within expected tolerance both on display and in the iOS app:

![image](https://user-images.githubusercontent.com/12213/234027269-87f0d015-4dd6-4d1e-b905-7360a6f34bf7.jpeg)

![image](https://user-images.githubusercontent.com/12213/234027331-40aa0390-e140-42ff-84a8-5b1fe9aeab63.jpeg)

**Comment by jm3** [comment]
Wow — I changed nothing, opened Homebridge web UI to look for the requested logs, and now the Awair temps and humidity display within expected Fahrenheit (and humidity) ranges.

I changed exactly **nothing**(!) 🙃 

<img width="370" alt="Home app showing summary of Awair metrics" src="https://user-images.githubusercontent.com/12213/234037566-d1a48892-03bd-4dc8-b6a8-78cc07d481e2.png">
<img width="727" alt="Homebridge web app showing summary of Awair metrics" src="https://user-images.githubusercontent.com/12213/234037588-0dac1290-376b-40c2-82d1-00dd529e3369.png">

**Comment by jm3** [comment]
For anyone who comes after and just scrolls to the bottom to find the fix — my Awair wasn't reporting Celsius, but was instead returning no result — the 32º was a "zero" value (in Celsius!) correctly converted to Fahrenheit, because the device or Awair's cloud service was (unexpectedly) returning zero / no data. :)

**Comment by DMBlakeley** [comment]
@jm3, thanks for the additional clarification to watch out for "zero" values as an indication that data for your device is offline with the Awair cloud. This usually corrects after a few hours.
